public/app/features/plugins/admin/components/PluginDetailsBody.tsx

```diff
@@ -197,3 +199,20 @@ export const getStyles = (theme: GrafanaTheme2) => ({
     },
   }),
 });
+
+function shouldDisableInstallation(plugin: CatalogPlugin) {
```

Comment:
I wonder if we should move this function to a place where is reusable and easier to discover, if we plan to do more changes to the catalog and perhaps render install buttons somewhere else we might want to reuse this logic

Reply:
Agree with @academo. I tried to do this a while ago [here](https://github.com/grafana/grafana/pull/93205/files#diff-1ef1d0d94dcbc880c46d454f8d88490993391f1c87170a3782ffa3ed74eaff02R429), but not sure it's a 1:1. Feel free to play around with it if you see a way to re-use some logic